### PR TITLE
Change vaultVersion to protocolVersion

### LINKED ISSUE #330: Change vaultVersion to protocolVersion
In API vaultVersion has changed to protocolVersion (cowamm remains 1). We need to update SDK to reflect this.

### PR DESCRIPTION
**Pull request by johngrantuk**
Closes #330. I'm not 100% confident this doesn't break anything, may be worth waiting until we address #332 to make sure flow works with API.

### CODE REVIEW

**Comment by brunoguerios** [comment]
I agree that having #332 done should give us more confidence that this PR is ready to go ✅

File: src/entities/addLiquidity/index.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export class AddLiquidity implements AddLiquidityBase {
         poolState: PoolState,
     ): Promise<AddLiquidityQueryOutput> {
         this.inputValidator.validateAddLiquidity(input, poolState);
-        switch (poolState.vaultVersion) {
+        switch (poolState.protocolVersion) {
             case 0: {
```
Comment (gmbronco) [inline src/entities/addLiquidity/index.ts:26]:
```suggestion
            case 1: {
```

File: src/entities/addLiquidity/index.ts [hunk 2]
```diff
@@ -42,7 +42,7 @@ export class AddLiquidity implements AddLiquidityBase {
     }
 
     buildCall(input: AddLiquidityBuildCallInput): AddLiquidityBuildCallOutput {
-        switch (input.vaultVersion) {
+        switch (input.protocolVersion) {
             case 0: {
```
Comment (gmbronco) [inline src/entities/addLiquidity/index.ts:46]:
```suggestion
            case 1: {
```

File: src/entities/addLiquidity/types.ts [hunk 3]
```diff
@@ -48,7 +48,7 @@ export type AddLiquidityBaseQueryOutput = {
     amountsIn: TokenAmount[];
     chainId: number;
     tokenInIndex?: number;
-    vaultVersion: 0 | 2 | 3;
+    protocolVersion: 0 | 2 | 3;
```
Comment (gmbronco) [inline src/entities/addLiquidity/types.ts:51]:
```suggestion
    protocolVersion: 1 | 2 | 3;
```

File: src/entities/removeLiquidity/index.ts [hunk 4]
```diff
@@ -24,7 +24,7 @@ export class RemoveLiquidity implements RemoveLiquidityBase {
         poolState: PoolState,
     ): Promise<RemoveLiquidityQueryOutput> {
         this.inputValidator.validateRemoveLiquidity(input, poolState);
-        switch (poolState.vaultVersion) {
+        switch (poolState.protocolVersion) {
             case 0: {
```
Comment (gmbronco) [inline src/entities/removeLiquidity/index.ts:28]:
```suggestion
            case 1: {
```

File: src/entities/removeLiquidity/index.ts [hunk 5]
```diff
@@ -54,7 +54,7 @@ export class RemoveLiquidity implements RemoveLiquidityBase {
         poolState: PoolStateWithBalances,
     ): Promise<RemoveLiquidityQueryOutput> {
         this.inputValidator.validateRemoveLiquidityRecovery(input, poolState);
-        switch (poolState.vaultVersion) {
+        switch (poolState.protocolVersion) {
             case 0: {
```
Comment (gmbronco) [inline src/entities/removeLiquidity/index.ts:58]:
```suggestion
            case 1: {
```

File: src/entities/removeLiquidity/index.ts [hunk 6]
```diff
@@ -80,7 +80,7 @@ export class RemoveLiquidity implements RemoveLiquidityBase {
         input: RemoveLiquidityBuildCallInput,
     ): RemoveLiquidityBuildCallOutput {
         const isV2Input = 'sender' in input;
-        switch (input.vaultVersion) {
+        switch (input.protocolVersion) {
             case 0: {
```
Comment (gmbronco) [inline src/entities/removeLiquidity/index.ts:84]:
```suggestion
            case 1: {
```

File: test/cowAmm/addLiquidity.integration.test.ts [hunk 7]
```diff
@@ -36,7 +36,7 @@ import {
     TOKENS,
 } from 'test/lib/utils';
 
-const vaultVersion = 0;
+const protocolVersion = 0;
```
Comment (gmbronco) [inline test/cowAmm/addLiquidity.integration.test.ts:39]:
```suggestion
const protocolVersion = 1;
```

File: test/cowAmm/removeLiquidity.integration.test.ts [hunk 8]
```diff
@@ -42,7 +42,7 @@ import {
     TOKENS,
 } from 'test/lib/utils';
 
-const vaultVersion = 0;
+const protocolVersion = 0;
```
Comment (gmbronco) [inline test/cowAmm/removeLiquidity.integration.test.ts:45]:
```suggestion
const protocolVersion = 1;
```

File: test/lib/utils/addLiquidityHelper.ts [hunk 9]
```diff
@@ -310,7 +310,7 @@ export function assertAddLiquidityProportional(
     addLiquidityInput: AddLiquidityProportionalInput,
     addLiquidityOutput: AddLiquidityOutput,
     slippage: Slippage,
-    vaultVersion: 0 | 2 | 3 = 2,
+    protocolVersion: 0 | 2 | 3 = 2,
```
Comment (gmbronco) [inline test/lib/utils/addLiquidityHelper.ts:313]:
```suggestion
    protocolVersion: 1 | 2 | 3 = 2,
```

File: test/lib/utils/addLiquidityHelper.ts [hunk 10]
```diff
@@ -429,7 +429,7 @@ function assertAddLiquidityBuildCallOutput(
     addLiquidityBuildCallOutput: AddLiquidityBuildCallOutput,
     isExactIn: boolean,
     slippage: Slippage,
-    vaultVersion: 0 | 2 | 3 = 2,
+    protocolVersion: 0 | 2 | 3 = 2,
```
Comment (gmbronco) [inline test/lib/utils/addLiquidityHelper.ts:432]:
```suggestion
    protocolVersion: 1 | 2 | 3 = 2,
```

File: src/entities/removeLiquidity/types.ts [hunk 11]
```diff
@@ -63,7 +63,7 @@ export type RemoveLiquidityBaseQueryOutput = {
     bptIn: TokenAmount;
     amountsOut: TokenAmount[];
     tokenOutIndex?: number;
-    vaultVersion: 0 | 2 | 3;
+    protocolVersion: 0 | 2 | 3;
```
Comment (gmbronco) [inline src/entities/removeLiquidity/types.ts:66]:
```suggestion
    protocolVersion: 1 | 2 | 3;
```

File: src/entities/types.ts [hunk 12]
```diff
@@ -7,7 +7,7 @@ export type PoolState = {
     address: Address;
     type: string;
     tokens: MinimalToken[];
-    vaultVersion: 0 | 2 | 3;
+    protocolVersion: 0 | 2 | 3;
```
Comment (gmbronco) [inline src/entities/types.ts:10]:
```suggestion
    protocolVersion: 1 | 2 | 3;
```

File: src/entities/types.ts [hunk 13]
```diff
@@ -16,7 +16,7 @@ export type PoolStateWithBalances = {
     type: string;
     tokens: PoolTokenWithBalance[];
     totalShares: HumanAmount;
-    vaultVersion: 0 | 2 | 3;
+    protocolVersion: 0 | 2 | 3;
```
Comment (gmbronco) [inline src/entities/types.ts:19]:
```suggestion
    protocolVersion: 1 | 2 | 3;
```

File: src/entities/swap/paths/types.ts [hunk 14]
```diff
@@ -8,5 +8,5 @@ export type Path = {
     tokens: TokenApi[];
     outputAmountRaw: bigint;
     inputAmountRaw: bigint;
-    vaultVersion: 0 | 2 | 3;
+    protocolVersion: 0 | 2 | 3;
```
Comment (gmbronco) [inline src/entities/swap/paths/types.ts:11]:
```suggestion
    protocolVersion: 1 | 2 | 3;
```

File: src/entities/addLiquidity/addLiquidityCowAmm/index.ts [hunk 15]
```diff
@@ -51,7 +51,7 @@ export class AddLiquidityCowAmm implements AddLiquidityBase {
             amountsIn,
             tokenInIndex,
             chainId: input.chainId,
-            vaultVersion: 0,
+            protocolVersion: 0,
```
Comment (gmbronco) [inline src/entities/addLiquidity/addLiquidityCowAmm/index.ts:54]:
```suggestion
            protocolVersion: 1,
```

**Review by gmbronco — COMMENTED** [review]
To maintain consistency with SC versioning, we have updated the cowAmm protocolVersion from 0 to 1 in the API.

**Review by brunoguerios — APPROVED** [review]

**Pull request merged**